site/content/en/latest/design/security-policy.md

```diff
@@ -0,0 +1,114 @@
+---
+title: "SecurityPolicy "
+---
+
+## Overview
+
+This design document introduces the `SecurityPolicy` API allowing system administrators to configure
+authentication and authorization policies to the traffic entering the gateway.
+
+## Goals
+* Add an API definition to hold settings for configuring authentication and authorization rules
+on the traffic entering the gateway.
+
+## Non Goals
+* Define the API configuration fields in this API.
+
+## Implementation
+`SecurityPolicy` is a [Policy Attachment][] type API that can be used to extend [Gateway API][]
+to define authentication and authorization rules..
+
+### Example
+Here is an example highlighting how a user can configure this API.
+
+```
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: GatewayClass
+metadata:
+  name: eg
+spec:
+  controllerName: gateway.envoyproxy.io/gatewayclass-controller
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: eg
+  namespace: default
+spec:
+  gatewayClassName: eg
+  listeners:
+    - name: https
+      protocol: HTTPS
+      port: 443
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: backend
+  namespace: default
+spec:
+  parentRefs:
+    - name: eg
+  hostnames:
+    - "www.example.com"
+  rules:
+    - backendRefs:
+        - group: ""
+          kind: Service
+          name: backend
+          port: 3000
+          weight: 1
+      matches:
+        - path:
+            type: PathPrefix
+            value: /
+---
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: SecurityPolicy
+metadata:
+  name: jwt-authn-policy
+  namespace: default
+spec:
+  jwt:
+    providers:
+    - name: example
+      remoteJWKS:
+        uri: https://raw.githubusercontent.com/envoyproxy/gateway/main/examples/kubernetes/authn/jwks.json
+  targetRef:
+    group: gateway.networking.k8s.io
+    kind: Gateway
+    name: eg
+    namespace: default
+```
+
+## Features / API Fields
+Here is a list of features that can be included in this API
+* JWT based authentication
+* OIDC Authentication
+* External Authorization
+* Basic Auth
+* API Key Auth
+* CORS
+
+## Design Decisions
+* This API will only support a single `targetRef` and can bind to a `Gateway` resource or a `HTTPRoute` or `GRPCRoute`.
+* This API resource MUST be part of same namespace as the targetRef resource
+* There can be only be ONE policy resource attached to a specific targetRef e.g. a `Listener` (section)  within a `Gateway`
+* If the policy targets a resource but cannot attach to it, this information should be reflected
```

Comment:
Is this the solution for OIDC before per-route configuration is supported?  https://github.com/envoyproxy/envoy/issues/29641

Reply:
we cannot add OIDC into this API until per route is supported

Reply:
Adding Per-route configuration for OIDC is a big change, and it won't be available soon. How about we start by providing OIDC at the Gateway and Listener levels and then add per-route support when it's ready?   Listener-level OIDC should cover most of what EG users need.

Per-route OIDC issue: https://github.com/envoyproxy/envoy/issues/29641

Reply:
ptal @envoyproxy/gateway-maintainers

Reply:
I verified that per-route configuration can be accomplished with "route level filter disabled support", so this is no longer blocking us. https://github.com/zhaohuabing/playground/blob/main/envoy/per-route-oauth2-oidc/envoy.yaml